Use `CMAKE_CURRENT_LIST_DIR` instead of CMAKE_CURRENT_SOURCE_DIR in CMakeLists.txt

Currently, this project uses:

```
project(Vc VERSION "${CMAKE_MATCH_1}.${CMAKE_MATCH_2}.${CMAKE_MATCH_3}" LANGUAGES C CXX)
list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
```

to add the module directory into CMAKE_MODULE_PATH. However, it doesn't behave as expected when users try to embed this project into their own using CPM.cmake:

```
CMake Error at /home/lz/.cache/CPM/vc/c43a073d69c63625768cfeb76b77ea806113471c/CMakeLists.txt:25 (include):
  include could not find requested file:

    VcMacros


CMake Error at /home/lz/.cache/CPM/vc/c43a073d69c63625768cfeb76b77ea806113471c/CMakeLists.txt:26 (include):
  include could not find requested file:

    AddTargetProperty


CMake Error at /home/lz/.cache/CPM/vc/c43a073d69c63625768cfeb76b77ea806113471c/CMakeLists.txt:27 (include):
  include could not find requested file:

    OptimizeForArchitecture


CMake Error at /home/lz/.cache/CPM/vc/c43a073d69c63625768cfeb76b77ea806113471c/CMakeLists.txt:29 (vc_determine_compiler):
  Unknown CMake command "vc_determine_compiler"
```

This is because CMAKE_CURRENT_SOURCE_DIR only refers to the parent project's source dir when this project is embedded, which can be fixed with [`CMAKE_CURRENT_LIST_DIR` which points to the correct directory](https://stackoverflow.com/questions/15662497/difference-between-cmake-current-source-dir-and-cmake-current-list-dir)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@ file(STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/Vc/version.h _version_lines REGEX "^#de
 string(REGEX MATCH "([0-9]+)\\.([0-9]+)\\.([0-9]+)" _version_matches "${_version_lines}")
 
 project(Vc VERSION "${CMAKE_MATCH_1}.${CMAKE_MATCH_2}.${CMAKE_MATCH_3}" LANGUAGES C CXX)
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
 set(disabled_targets)
 

